File: crypto_and_hash_cracking/files/Challenges/tests/LCG-check.java

package lastegg;

import java.math.BigInteger;

public class LCG {

public static void main(String args[]) {
BigInteger n = new BigInteger("2073867623");//seed
BigInteger m = new BigInteger("2147483648");//modulus
BigInteger a = new BigInteger("13371337");//multipler
BigInteger c = new BigInteger("1015244223");//increment

n = (a.multiply(n).add(c)).mod(m);

for (int i = 0; i < 10; i++) {
System.out.println(n.toString().substring(n.toString().length()-6));
n = (a.multiply(n).add(c)).mod(m);

}
System.out.println("... ");
}
}

File: crypto_and_hash_cracking/files/Challenges/tests/LCG - check.py
a = 1357172052#Multiplier
c = 1304548919 	#Increment
m = 2147483648#Modulus
xi = 1240717431 #Seed

xi = (a*xi + c)%m

for i in range(10):
print str(xi)[-6:]
xi = (a*xi + c)%m

print "done!"
